examples/dreamer/dreamer.py

```diff
@@ -0,0 +1,493 @@
+import dataclasses
```

Comment:
I took a quick pass at the code, and for now i only have two high level comments. The first one is minor, but the second I think is important:

- This PR is really large, and from what I can see it has a lot of independent small features that could have been merged as separate PRs. 
- There is a lot of boilerplate code in `dreamer.py` that makes it hard to follow the logic easily. I would imagine that one of the main goals of this module is to illustrate how to use TorchRL components to write sophisticated algorithms (is this correct?). However, there is a lot of code here for setting configurations, video recording, logging, etc., all intermingled with the RL code, and I argue this reduces the mental space that a newcomer will have for recognizing the TorchRL features that are enabling this Dreamer implementation. For me, the closer this file would look to readable pseudocode, the better job it'd be doing at selling the library. 

My suggestion is then to move a lot of this boilerplate to utility modules, imported with easily interpretable function names, and also adding a lot of comments in the code guiding users through the important features used in the implementation.